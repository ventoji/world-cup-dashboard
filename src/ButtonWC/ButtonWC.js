import React from 'react';
import PropTypes from 'prop-types';

/** Defautl button WC  */
const ButtonWC = ({ label, onClick, ...props }) => {
  return (
    <button type="submit" onClick={onClick} {...props}>
      {label}
    </button>
  );
};

ButtonWC.propTypes = {
  /** Button Contents */
  label: PropTypes.string.isRequired,

  /** Function to call onClick */
  onClick: PropTypes.func.isRequired,
};

ButtonWC.defaultProps = {
  label: 'start game',
  onClick: () => {},
};

export default ButtonWC;
