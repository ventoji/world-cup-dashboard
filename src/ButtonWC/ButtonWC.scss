.wc-button {
  padding: 3px 3px;
}
