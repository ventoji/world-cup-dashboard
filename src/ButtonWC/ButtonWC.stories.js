import React from 'react';
import { withKnobs } from '@storybook/addon-knobs';
import ButtonWC from './ButtonWC';

export default {
  title: 'WC Components/ButtonWC',
  component: ButtonWC,
  decorators: [withKnobs],
};

const Template = (args) => <ButtonWC {...args} />;

export const Default = Template.bind({});
