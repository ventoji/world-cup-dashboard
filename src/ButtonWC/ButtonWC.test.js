import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ButtonWC from './ButtonWC';

describe('Button component', () => {
  let button; // use shallow

  beforeEach(() => {
    button = shallow(<ButtonWC label="button test" onClick={() => {}} />);
  });

  it('should be defined', () => {
    expect(ButtonWC).toBeDefined();
  });

  it('should be submit type', () => {
    expect(toJson(button)).toMatchSnapshot();

    expect(button.find({ type: 'submit' })).toBeTruthy();
  });

  it('should call mock function when button is clicked', () => {
    const mockHandleClick = jest.fn();

    const button = shallow(<ButtonWC onClick={mockHandleClick} label="Ok" />);

    // button.find('button').simulate('click');
    button.simulate('click');

    expect(mockHandleClick.mock.calls.length).toEqual(1);
    expect(mockHandleClick).toHaveBeenCalled();
  });
});
