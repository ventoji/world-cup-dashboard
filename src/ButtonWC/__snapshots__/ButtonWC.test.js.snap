// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should be submit type 1`] = `
<button
  className="wc-button"
  onClick={[Function]}
  type="submit"
>
  button test
</button>
`;
